src/AutoRest.CSharp/LowLevel/Generation/LowLevelModelWriter.cs

```diff
@@ -2,40 +2,17 @@
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
 using System;
-using System.Collections.Generic;
-using System.Collections.Immutable;
 using System.Linq;
-using System.Text;
-using AutoRest.CSharp.Generation.Writers;
+using System.Text.Json;
 using AutoRest.CSharp.Output.Models;
 using AutoRest.CSharp.Output.Models.Types;
-using AutoRest.CSharp.Utilities;
-using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Azure.Core;
 
 namespace AutoRest.CSharp.Generation.Writers
 {
     internal class LowLevelModelWriter
     {
-        public static void WriteModel(CodeWriter writer, TypeProvider model)
```

Comment:
@archerzz I have removed the input type pattern matching because I don't expect we will need it. CADL has models and enums as separate collections (https://github.com/microsoft/cadl/blob/main/packages/compiler/core/types.ts#L275), and I think we should follow this pattern.